Make `EmbedsRelations::embedsMany` and `EmbedsRelations::embedsOne` public

GromNaN opened the pull request:
Fix #2555

Other relation methods are public.

GromNaN committed:
Make EmbedsRelations::embedsMany and EmbedsRelations::embedsOne public

alcaeus commented on src/Eloquent/EmbedsRelations.php:
```suggestion
     * Define an embedded one-to-one relationship.
```

alcaeus reviewed (commented):
Changes LGTM, and while I think we can skip test updates, fixing the description of `embedsOne` would be good.

GromNaN committed:
Fix docblock for embedsOne and add tests for public visibility

- Fix embedsOne docblock: "one-to-many" -> "one-to-one"
- Add tests for direct call and resolveRelationUsing use case
